src/nanoarrow/array.c

```diff
@@ -168,21 +168,40 @@ ArrowErrorCode ArrowArrayInitFromType(struct ArrowArray* array,
 ArrowErrorCode ArrowArrayInitFromArrayView(struct ArrowArray* array,
                                            struct ArrowArrayView* array_view,
                                            struct ArrowError* error) {
-  ArrowArrayInitFromType(array, array_view->storage_type);
+  NANOARROW_RETURN_NOT_OK_WITH_ERROR(
+      ArrowArrayInitFromType(array, array_view->storage_type), error);
```

Comment:
Seems like you could guard against forgetting with `-Wunused-result` and manually annotating functions with the (compiler specific) `__attribute__((warn_unused_result))` though that is rather a hassle